Mathlib/Tactic/Linter/Style.lean

```diff
@@ -68,4 +69,47 @@ initialize addLinter setOptionLinter
 
 end Style.setOption
 
+/-- The `check_declID` linter: if it emits a warning, then a declID is considered
+non-standard style.  -/
+register_option linter.style.check_declID : Bool := {
+  defValue := false
+  descr := "enable the `setOption` linter"
+}
+
+
+namespace Style.checkDeclID
+
+/-- Checks whether a given identifier name contains "__". -/
+def contains_double_underscore (stx : Syntax) : Bool :=
+  match stx.getSubstring? with
+  | some str => 1 < (str.toString.splitOn "__").length
+  | none => false
+
+/-- The `checkDeclID` linter: if this linter emits a warning, then a declID is considered
+non-standard style. Currently we only check if it contains a double underscore ("__") as a
+substring.
+
+**Why is this bad?** Double underscores in theorem names can be considered non-standard style and
+probably have been introduced by accident
+**How to fix this?** Use single underscores to separate parts of a name, following standard naming
+conventions.
+-/
+def checkDeclIDLinter: Linter where
+  run := withSetOptionIn fun _stx => do
+    unless Linter.getLinterValue linter.style.check_declID (← getOptions) do
+      return
+    if (← MonadState.get).messages.hasErrors then
+      return
+    for stx in (← getNames _stx) do
+      if (contains_double_underscore stx) then
```

Comment:
I think mathlib style is to not brace if conditions.

Reply:
removed the braces 
(I have written too much C++ recently...)